### PR TITLE
DCWL-2489 | Scala 3 upgrade

### PR DESCRIPTION
**Pull request by Jack-Brannon**
Please ensure we are following our Ways of Working for Pull Requests: https://confluence.tools.tax.service.gov.uk/display/BTL/Ways+of+Working+for+Pull+Requests

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -7,9 +7,6 @@ addSbtPlugin("com.github.sbt"    %  "sbt-release"           % "1.0.15")
...
 addSbtPlugin("uk.gov.hmrc"       %  "sbt-distributables"    % "2.5.0")
 addSbtPlugin("net.virtual-void"  %  "sbt-dependency-graph"  % "0.10.0-RC1")
-addSbtPlugin("org.scoverage"     %  "sbt-scoverage"         % "2.0.12")
-addSbtPlugin("org.scalastyle"    %% "scalastyle-sbt-plugin" % "1.0.0" exclude("org.scala-lang.modules", "scala-xml_2.12"))
-addSbtPlugin("com.timushev.sbt"  %  "sbt-updates"           % "0.6.3")
-// To resolve a bug with version 2.x.x of the scoverage plugin - https://github.com/sbt/sbt/issues/6997
-// Try to remove when sbt 1.8.0+ and scoverage is 2.0.7+
+addSbtPlugin("org.scoverage"     %  "sbt-scoverage"         % "2.2.1")
+
 ThisBuild / libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always
```
Comment (gavinharris43) [inline project/plugins.sbt:12]: Did you mean to remove this

File: project/plugins.sbt [hunk 2]
```diff
@@ -7,9 +7,6 @@ addSbtPlugin("com.github.sbt"    %  "sbt-release"           % "1.0.15")
 addSbtPlugin("org.playframework" %  "sbt-plugin"            % "3.0.3")
 addSbtPlugin("uk.gov.hmrc"       %  "sbt-distributables"    % "2.5.0")
 addSbtPlugin("net.virtual-void"  %  "sbt-dependency-graph"  % "0.10.0-RC1")
-addSbtPlugin("org.scoverage"     %  "sbt-scoverage"         % "2.0.12")
-addSbtPlugin("org.scalastyle"    %% "scalastyle-sbt-plugin" % "1.0.0" exclude("org.scala-lang.modules", "scala-xml_2.12"))
-addSbtPlugin("com.timushev.sbt"  %  "sbt-updates"           % "0.6.3")
```
Comment (gavinharris43) [inline project/plugins.sbt:12]: Isnt this for dependency updates

File: run_all_tests.sh [hunk 3]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
 
-sbt clean scalastyle coverage test it:test coverageReport dependencyUpdates
+sbt clean coverage test it:test coverageReport dependencyUpdate
```
Comment (gavinharris43) [inline run_all_tests.sh:3]: "sbt-updates"

File: test/component/Utf8Spec.scala [hunk 4]
```diff
@@ -60,9 +60,10 @@ class Utf8Spec extends ComponentTestSpec with AuditService with ExpectedTestResp
     stopMockServer()
   }
 
-  val ValidRawXmlByte = ByteString.fromArray(Files.readAllBytes(Paths.get("target/scala-2.13/test-classes/raw/valid_xml.raw")))
-  val InvalidRawXmlByte = ByteString.fromArray(Files.readAllBytes(Paths.get("target/scala-2.13/test-classes/raw/invalid_xml.raw")))
-
+  // TODO: get Scala version number, avoid string literal
```
Comment (gavinharris43) [inline test/component/Utf8Spec.scala:63]: Todo?

**Review by gavinharris43 — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/customs/declaration/controllers/ErrorResponses.scala [hunk 5]
```diff
@@ -86,7 +86,7 @@ case class ErrorResponse(httpStatusCode: Int, errorCode: String, message: String
     }
 
   def withErrors(contents: ResponseContents*): ErrorResponse = {
-    ErrorResponse(this.httpStatusCode, this.errorCode, this.message, contents :_ *)
```
[the remark this answers is not in the document]
Comment (Jack-Brannon) [reply app/uk/gov/hmrc/customs/declaration/controllers/ErrorResponses.scala:89]: The syntax `x: _*` is no longer supported for vararg splices; use `x*` instead

File: app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/payloadValidationActions.scala [hunk 6]
```diff
@@ -105,7 +105,7 @@ abstract class PayloadValidationAction(val xmlValidationService: XmlValidationSe
         case saxe: SAXException =>
           val msg = "Payload is not valid according to schema"
           logger.debug(s"$msg:\n${xml.toString()}", saxe)
-          Left(errorResponse(msg, xml.toString(), xmlValidationErrors(saxe): _*))
```
[the remark this answers is not in the document]
Comment (Jack-Brannon) [reply app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/payloadValidationActions.scala:108]: +1

File: app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/upscan/fileUploadPayloadValidationActions.scala [hunk 7]
```diff
@@ -116,44 +116,44 @@ class FileUploadPayloadValidationComposedAction @Inject()(val fileUploadPayloadV
...
     def fileSequenceNoLessThanOne = validate(
       fileUpload,
-      { b: FileUploadRequest =>
+      { (b: FileUploadRequest) =>
         b.files.head.fileSequenceNo.value == 1},
       errorFileSequenceNoLessThanOne)
     errorRedirectOnly ++ maxFileGroupSize ++ maxFileSequenceNo ++ fileGroupSize ++ duplicateFileSequenceNo ++ fileSequenceNoLessThanOne match {
       case Seq() => Right(())
       case errors =>
-        Left(new ErrorResponse(Status.BAD_REQUEST, BadRequestCode, "Payload did not pass validation", errors: _*))
```
[the remark this answers is not in the document]
Comment (Jack-Brannon) [reply app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/upscan/fileUploadPayloadValidationActions.scala:156]: +1

File: project/plugins.sbt (hunk 2 above)
Comment (Jack-Brannon) [reply project/plugins.sbt:12]: Yep removed wrong line. Added sbt-updates back in. and removed scala-xml.

File: run_all_tests.sh (hunk 3 above)
Comment (Jack-Brannon) [reply run_all_tests.sh:3]: Added sbt-updates back and undid this change. Tested and get updates report back

File: test/component/Utf8Spec.scala (hunk 4 above)
Comment (Jack-Brannon) [reply test/component/Utf8Spec.scala]: Completed ToDo, added plugin and config to allow us to fetch build info dynamically in the future.

**Review by gavinharris43 — APPROVED** [review]

**Pull request merged**